### ISSUE TITLE
a difference between the result in sklearn and in statsmodels

### ISSUE THREAD

**Issue by alienx-zero** [statsmodels/statsmodels#7457]
#### Describe the bug
I try to use cca to analyze generated data. But the weight from CCA in sklearn is different with the weight from CCA in statsmodels. Scaling is not a problem but why the negative weight in sklearn is positive in statsmodels. The sign did not make influence to advanced analysis?

here is the code
#### Code Sample, a copy-pastable example if possible

import numpy as np
import matplotlib.pyplot as plt
from sklearn.cross_decomposition import PLSCanonical, PLSRegression, CCA
from statsmodels.multivariate.cancorr import CanCorr
from numpy.linalg import svd
np.random.seed(1)

n = 500
l1 = np.random.normal(size=n)
l2 = np.random.normal(size=n)

latents = np.array([l1, l1, l2, l2]).T
latents_x = np.array([l1, l1, l2, l2, l2]).T
X = latents_x + np.random.normal(size=5 * n).reshape((n, 5))
Y = latents + np.random.normal(size=4 * n).reshape((n, 4))

X_train = X[:n // 2]
Y_train = Y[:n // 2]
X_test = X[n // 2:]
Y_test = Y[n // 2:]

print("Corr(X)")
print(np.round(np.corrcoef(X.T), 2))
print("Corr(Y)")
print(np.round(np.corrcoef(Y.T), 2))

def print_weights(name, weights):
    first = weights[:, 0]
    print(name + ': ' + ', '.join(['{:.3f}'.format(item) for item in first / np.max(first)]))
    
stats_cca = CanCorr(Y_train, X_train)

print(stats_cca.corr_test().summary())
X_train_c = stats_cca.x_cancoef
Y_train_c = stats_cca.y_cancoef

cca = CCA(n_components=4)
cca.fit(X_train, Y_train)
X_train_r1, Y_train_r1 = cca.transform(X_train, Y_train)
X_train_r1w = cca.x_weights_
Y_train_r1w = cca.y_weights_

#### result
X_train_c
Out[18]: 
array([[ 7.96303132e-03, -3.12871021e-02, -1.25393423e-02, 1.20338231e-02],
       [ 3.10614617e-03, -2.32485643e-02,  1.44761636e-02, -1.75946328e-02],
       [-1.32660099e-02, -9.09194463e-05, -3.89927192e-02, -4.21746862e-02],
       [-2.07297078e-02, -4.38177487e-03,  4.36532401e-02, -1.91784700e-04],
       [-1.77065606e-02, -5.39899163e-03, -1.46066622e-02, 4.30796190e-02]])
X_train_r1w
Out[19]: 
array([[-0.23636351, -0.23636351, -0.23636351, -0.23636351],
       [-0.09633587, -0.09633587, -0.09633587, -0.09633587],
       [ 0.39680365,  0.39680365,  0.39680365,  0.39680365],
       [ 0.68919865,  0.68919865,  0.68919865,  0.68919865],
       [ 0.54991251,  0.54991251,  0.54991251,  0.54991251]])

**Comment by alienx-zero** [comment]
I get the reason...
